wire-ios-data-model/Tests/UseCases/AccountImageSource/GetUserAccountImageSourceUseCaseTests.swift

```diff
@@ -0,0 +1,126 @@
+//
+
+import WireDataModelSupport
+import XCTest
+
+@testable import WireDataModel
+
+final class GetUserAccountImageSourceUseCaseTests: XCTestCase {
+
+    private var coreDataStack: CoreDataStack!
+    private var sut: GetUserAccountImageSourceUseCase!
+
+    @MainActor
+    override func setUp() async throws {
+        coreDataStack = try await CoreDataStackHelper().createStack()
+        sut = .init()
+    }
+
+    override func tearDown() {
+        sut = nil
+        coreDataStack = nil
+    }
+
+    func testAccountImage() async throws {
+        // Given
+        let accountImageData = try imageData(from: .brown)
+        coreDataStack.account.imageData = accountImageData
+        let user = await coreDataStack.viewContext.perform { [self] in
+            ZMUser.selfUser(in: coreDataStack.viewContext)
+        }
+
+        // When
+        let accountImageSource = try await sut.invoke(
+            user: user,
+            userContext: user.managedObjectContext,
+            account: coreDataStack.account
+        )
+
+        // Then
+        guard case .image(let accountImage) = accountImageSource, accountImage.pngData() == accountImageData else {
+            return XCTFail("Expected account image to match actual image")
+        }
+    }
+
+    func testUserInitials() async throws {
+        // Given
+        let user = await coreDataStack.viewContext.perform { [self] in
+            let user = ZMUser.selfUser(in: coreDataStack.viewContext)
+            user.name = " Wire\tUser \t\n"
+            return user
+        }
+
+        // When
+        let accountImageSource = try await sut.invoke(
+            user: user,
+            userContext: user.managedObjectContext,
+            account: coreDataStack.account
+        )
+
+        // Then
+        XCTAssertEqual(accountImageSource, .text("WU"))
+    }
+
+    func testAccountName() async throws {
+        // Given
+        coreDataStack.account.userName = "Wire\tUser \t\n"
+        let user = await coreDataStack.viewContext.perform { [self] in
+            ZMUser.selfUser(in: coreDataStack.viewContext)
+        }
+
+        // When
+        let accountImageSource = try await sut.invoke(
+            user: user,
+            userContext: nil,
+            account: coreDataStack.account
+        )
+
+        // Then
+        XCTAssertEqual(accountImageSource, .text("WU"))
+    }
+
+    func testNoSource() async throws {
+        // Given
+        let user = await coreDataStack.viewContext.perform { [self] in
+            ZMUser.selfUser(in: coreDataStack.viewContext)
+        }
+
+        do {
+            // When
+            _ = try await sut.invoke(
+                user: user,
+                userContext: user.managedObjectContext,
+                account: coreDataStack.account
+            )
+            XCTFail("Unexpected success")
+        } catch GetUserAccountImageSourceUseCase.Error.invalidImageSource {
+            // Then
+        }
```

Comment:
Same comment as in wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift